include/sleipnir/control/OCPSolver.hpp

```diff
@@ -0,0 +1,76 @@
+
+
+#pragma once
+
+#include <Eigen/Core>
+
+#include "sleipnir/SymbolExports.hpp"
+#include "sleipnir/optimization/OptimizationProblem.hpp"
+#include "sleipnir/autodiff/VariableMatrix.hpp"
+
+namespace sleipnir {
+
+typedef std::function<VariableMatrix(double, VariableMatrix, VariableMatrix)> FixedStepODE;
+typedef std::function<VariableMatrix(VariableMatrix, VariableMatrix)> FixedStepObjective;
+typedef std::function<void(VariableMatrix, VariableMatrix, bool)> FixedStepConstraintFunction;
+
+/**
+ * Performs 4th order Runge-Kutta integration of dx/dt = f(t, x, u) for dt.
+ *
+ * @param f  The function to integrate. It must take two arguments x and u.
+ * @param x  The initial value of x.
+ * @param u  The value u held constant over the integration period.
+ * @param t0 The initial time.
+ * @param dt The time over which to integrate.
+ */
+template <typename T, typename U, typename V>
+T RK4(std::function<T(V,T,U)> f, T x, U u, V t0, V dt) {
```

Comment:
Taking std::function by value here incurs a heap allocation due to type erasure and a potential copy. Consider using a template instead, like this:
```cpp
template <typename F, typename T, typename U, typename V>
T RK4(F&& f, T x, U u, V t0, V dt) {
```

We may want to consider more expressive template argument names like `template <typename F, typename State, typename Input, typename Time>`.